packages/ai-mcp-server/src/node/mcp-theia-server-impl.ts

```diff
@@ -0,0 +1,242 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { injectable, inject } from '@theia/core/shared/inversify';
+import { ILogger } from '@theia/core/lib/common/logger';
+import { BackendApplicationContribution } from '@theia/core/lib/node/backend-application';
+import { McpServer } from '@modelcontextprotocol/sdk/server/mcp';
+import { generateUuid } from '@theia/core';
+
+import { StreamableHTTPServerTransport } from '@modelcontextprotocol/sdk/server/streamableHttp';
+import * as express from '@theia/core/shared/express';
+import { randomUUID } from 'crypto';
+import { MCPTheiaServer, MCPServerConfig } from './mcp-theia-server';
+import { MCPBackendContributionManager } from './mcp-backend-contribution-manager';
+import { MCPFrontendContributionManager } from './mcp-frontend-contribution-manager';
+
+@injectable()
+export class MCPTheiaServerImpl implements MCPTheiaServer, BackendApplicationContribution {
+
+    @inject(ILogger)
+    protected readonly logger: ILogger;
+
+    @inject(MCPBackendContributionManager)
+    protected readonly backendContributionManager: MCPBackendContributionManager;
+
+    @inject(MCPFrontendContributionManager)
+    protected readonly frontendContributionManager: MCPFrontendContributionManager;
+
+    protected server?: McpServer;
+    protected httpTransports: Map<string, StreamableHTTPServerTransport> = new Map();
+    protected httpApp?: express.Application;
+    protected running = false;
+    protected config?: MCPServerConfig;
+    protected serverId: string = generateUuid();
+
+    async configure?(app: express.Application): Promise<void> {
+        this.httpApp = app;
+
+        try {
+            await this.start();
+        } catch (error) {
+            this.logger.error('Failed to start MCP server during initialization:', error);
+        }
+    }
```

Comment:
The MCP server is now always started without checking the config or the environment variable. Is that on purpose?

I'm fine with that behavior but then we should remove the config and adapt the Readmes accordingly.